internal/grpc/services/gateway/storageprovider.go

```diff
@@ -104,10 +104,29 @@ func (s *svc) getHome(ctx context.Context) string {
 	return "/home"
 }
 func (s *svc) InitiateFileDownload(ctx context.Context, req *provider.InitiateFileDownloadRequest) (*gateway.InitiateFileDownloadResponse, error) {
-	p, err := s.getPath(ctx, req.Ref)
+	statReq := &provider.StatRequest{Ref: req.Ref}
```

Comment:
We can't stat all download requests as these would fail for shared resources. We stat the resource directly only if `inSharedFolder` is false, the other requests are handled separately by checking for the conditions `isShareName` and `isShareChild`.

Reply:
^^ this got caught up in translation. Started using Goland recently and using git on the UI sometimes confuses me. Should be addressed on the latest commit :) good eye, thanks